src/controllers/dlgprefcontroller.cpp

```diff
@@ -45,6 +45,7 @@ DlgPrefController::DlgPrefController(QWidget* parent,
           m_pOutputTableModel(nullptr),
           m_pOutputProxyModel(nullptr),
           m_GuiInitialized(false),
+          m_inSlotPresetSelected(false),
```

Comment:
I'm not happy with this. Its essentially used for synchronization. If the respective methods were called from different threads, it was subject to race conditions. Its probably not called from different threads, qt doesn't tell us (probably for the better), but this is still a less than ideal hack. 

From what I can tell, this only required to change the behavior of `savePreset` while the "Mapping has been edited"-messagebox is showing, correct? Can we come up with a more reasonable (solution thats easier to understand later) solution?

Reply:
Alright, I get your point.
Yes, this was a rather lazy fix I admit.

The issue with a dirty mapping and a newly selected mapping is that `savePreset` calls `enumeratePresets` which also reselects the configured mapping or the presetPath it is called with (which triggers `slotPresetSelected`)

A better _workaround_ would be to restore the new selection in `slotPresetSelected` after saving the dirty previous mapping. (though without refactoring we'd still have that unnecessary first call of `slotPresetSelected`).
I'll do that.

The actual, comprehensive _fix_ would be to extract the preset selection from `enumeratePresets` into a separate function, but tbh I'm not up to this kind of refactoring for this bugfix.

Reply:
Mhmm I understand. What do you think about reverting that commit for now and retrying some other time? Because the rest of the PR looks good IMO. 

Also please don't get me wrong. I didn't mean to insult your effort or anything. I'm grateful for it. I'm sorry if seemed that way.

Reply:
No worries, your objection is reasonable.
I don't want to retry another time, because a) I don't want to just leave that UI inconsistentcy and b) I have too many unfinished PRs already so I want to finish the controller pref fixes asap.

I reverted that commit and changed `void savePreset()` to `bool savePreset`, extracted `enumeratePresets(presetPath)` from there and call it explicitely if required.

Reply:
I can't picture that solution in my head. Can you push the commit?

Reply:
I just did
[057ad01](https://github.com/mixxxdj/mixxx/pull/10821/commits/057ad017e9aeffdd14f92ad0f8cfc1d7d1a23ae2)

Reply:
Thanks. Much better.